book_syr_drought/01_asi.qmd

```diff
@@ -0,0 +1,200 @@
+    q_rank <- rank / (length(x) + 1)
+    rp <- 1 / q_rank
+    return(rp)
+}
+```
+
+```{r}
+box::use(
+    cumulus
+)
+box::use(
+    lubridate[...],
+    dplyr[...],
+    tidyr[...],
+    readr[...],
+    ggplot2[...],
+    glue[...],
+    forcats[...],
+    gghdx[...],
+    sf[...]
+)
+df_asi <- cumulus$fao_asi_adm1_tabular(iso3 = "syr")
+gdf_syr <- cumulus$download_fieldmaps_sf(iso3 = "syr", layer = "syr_adm1")
+
+gghdx()
+```
+
+```{r}
+df_filtered <- df_asi |>
+    mutate(
+        mo_num = as.numeric(Month),
+        harvest_year = case_when(
+            mo_num %in% c(11, 12) ~ Year + 1,
+            TRUE ~ Year
+        ),
+        mo_chr = fct_expand(month.abb[mo_num], month.abb),
+        date = glue("{Year}-{Month}-21")
+    ) |>
+    filter(
+        Dekad == 3,
+        mo_num %in% c(11, 12, 1, 2, 3, 4)
+    ) |>
+    mutate(
+        mo_fct = fct_relevel(mo_chr, c("Nov", "Dec", "Jan", "Feb", "Mar", "Apr")),
+        harvest_year_label = case_when(
+            harvest_year %in% c(2023:2025) ~ as.character(harvest_year),
+            .default = "Other"
+        ),
+        alpha_flag = if_else(harvest_year_label == "Other", "low", "high") # for alpha mapping
+    )
+
+df_rps <- df_filtered |>
+    group_by(ADM1_CODE, Province, mo_num, mo_chr) |>
+    mutate(
+        rp_empirical = rp_empirical(
+            Data,
+            ties_method = "average",
+            direction = "-1"
+        )
+    ) |>
+    arrange(
+        Province, mo_chr
+    )
+
+
+df_rps_2025 <- df_rps |>
+    filter(
+        harvest_year == 2025
+    )
+```
+
+## ASI Over the Season
+
+The numbers represent the return period of the ASI values for the 2025 harvest year (calculated empirically).
+
+```{r}
+df_filtered |>
+    mutate(
+        mo_fct = fct_relevel(mo_chr, c("Nov", "Dec", "Jan", "Feb", "Mar", "Apr")),
+        harvest_year_label = case_when(
+            harvest_year %in% c(2023:2025) ~ as.character(harvest_year),
+            .default = "Other"
+        ),
+        alpha_flag = if_else(harvest_year_label == "Other", "low", "high") # for alpha mapping
+    ) |>
+    ggplot(
+        aes(
+            x = mo_fct,
+            y = Data,
+            group = harvest_year,
+            color = harvest_year_label,
+            alpha = alpha_flag
+        )
+    ) +
+    geom_text(
+        data = df_rps_2025,
+        aes(label = scales::number(rp_empirical, accuracy = 1), vjust = -.5),
+        show.legend = FALSE
+    ) +
+    expand_limits(y = max(df_filtered$Data, na.rm = TRUE) * 1.3) +
+    labs(color = "Harvest Year", y = "ASI") +
+    scale_alpha_manual(values = c("low" = 0.3, "high" = 1), guide = "none") +
+    geom_line() +
+    facet_wrap(~Province) +
+    theme(
+        axis.text.x = element_text(angle = 90),
+        axis.title.x = element_blank()
+    )
+
+```
+
+## Map
+```{r}
+#| warning: false
+#| fig.height: 7
+
+gdf_syr_adm1 <- gdf_syr$syr_adm1 |>
+    mutate(
+        adm1_harm = case_when(
```

Comment:
Ah ok this explains why I was confused about the province names when I was looking at my dataframe vs your map. Where are these alternate names coming from?

Reply:
i think i was just getting the field map names to match w/ FAO ASI names. Could have gone the other way